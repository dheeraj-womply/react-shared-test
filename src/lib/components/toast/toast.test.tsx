import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { Toast } from './index'

test('Toast elements opened are OK', () => {
    const { container } = render(
        <Toast
            open={true}
            message="Login Success"
            onClose={() => {}}
        />
    )
    expect(container).toMatchSnapshot()
})

test('Toast elements closed are OK', () => {
    const { container } = render(
        <Toast
            open={false}
            message="Login Success"
            onClose={() => {}}
        />
    )
    expect(container).toMatchSnapshot()
})

test('Toast elements with action are OK', () => {
    const { container } = render(
        <Toast
            open={true}
            message="Login Success"
            action="Confirm"
            onClose={() => {}}
        />
    )
    expect(container).toMatchSnapshot()
})

test('Toast elements with alert props are OK', () => {
    const { container } = render(
        <Toast
            open={true}
            message="Login Failed"
            variant="outlined"
            severity="error"
            onClose={() => {}}
        />
    )
    expect(container).toMatchSnapshot()
})

test('Check if close function is called', async () => {
    const onClose = jest.fn()
    const { findByTestId } = render(
        <Toast
            open={true}
            message="Login Success"
            onClose={onClose}
        />
    )
    fireEvent.click(await findByTestId('toast-alert-action'))
    expect(onClose).toBeCalled()
})

test('Check if click away not trigger the close function', async () => {
    const onClose = jest.fn()
    const { findByTestId } = render(
        <Toast
            open={true}
            message="Login Success"
            onClose={onClose}
        />
    )
    fireEvent.click(document)
    expect(onClose).not.toBeCalled()
    expect(await findByTestId('toast-alert-action')).toBeInTheDocument()
})