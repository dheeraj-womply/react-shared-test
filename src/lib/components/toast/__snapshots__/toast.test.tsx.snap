// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast elements closed are OK 1`] = `<div />`;

exports[`Toast elements opened are OK 1`] = `
<div>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginTopLeft"
  >
    <div
      class="MuiPaper-root MuiAlert-root MuiAlert-filledInfo MuiPaper-elevation0"
      direction="down"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message"
      >
        Login Success
      </div>
      <div
        class="MuiAlert-action"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-colorInherit MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
          data-testid="toast-alert-action"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            OK
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Toast elements with action are OK 1`] = `
<div>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginTopLeft"
  >
    <div
      class="MuiPaper-root MuiAlert-root MuiAlert-filledInfo MuiPaper-elevation0"
      direction="down"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message"
      >
        Login Success
      </div>
      <div
        class="MuiAlert-action"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-colorInherit MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
          data-testid="toast-alert-action"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Confirm
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Toast elements with alert props are OK 1`] = `
<div>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginTopLeft"
  >
    <div
      class="MuiPaper-root MuiAlert-root MuiAlert-outlinedError MuiPaper-elevation0"
      direction="down"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message"
      >
        Login Failed
      </div>
      <div
        class="MuiAlert-action"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-colorInherit MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
          data-testid="toast-alert-action"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            OK
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
